docs: adapt Documentation with new docarray at center (#5941)
jina-ai/jina: docs/concepts/serving/executor/add-endpoints.md

The endpoint signature looks like the following:

```python
from typing import Dict, Union, Optional
from jina import Executor, requests
from docarray import DocList

class MyExecutor(Executor):
    @requests
    async def foo(
        self,

>         docs: DocList[...],

Reviewer: what is this `...` ?

Author: a place holder so that people understand is some generic

Reviewer: I think it might be confusing with the python `...` syntax. Maybe we could create a a dummy base doc for this ?

(streaming-endpoints)=
## Streaming endpoints

> Executors can stream Documents individually rather than as a whole DocLists. 

Reviewer 2: ```suggestion
Executors can stream Documents individually rather than as a whole DocList. 
```

Reviewer 2: or "whole DocLists" if that's more correct

> Jina offers a standard python client to use the streaming endpoint:

Reviewer 2: ```suggestion
Jina offers a standard python client for using the streaming endpoint:
```
